app/models/concerns/public_body_derived_fields.rb

```diff
@@ -0,0 +1,46 @@
+module PublicBodyDerivedFields
+    extend ActiveSupport::Concern
+
+    included do
+        before_save :set_first_letter
+
+        # When name or short name is changed, also change the url name
+        def short_name=(short_name)
```

Comment:
Looking at https://signalvnoise.com/posts/3372-put-chubby-models-on-a-diet-with-concerns and https://github.com/mysociety/alaveteli/blob/develop/app/models/concerns/translatable.rb, it looks like the `included` block is only needed for macros, so the method definitions should be outside of the block?